lib/variable/include/scipp/variable/reduction.h

```diff
@@ -46,12 +46,26 @@ namespace scipp::variable {
 [[nodiscard]] SCIPP_VARIABLE_EXPORT Variable nanmean(const Variable &var,
                                                      const Dim dim);
 
+// Reductions of all events within a bin.
+[[nodiscard]] SCIPP_VARIABLE_EXPORT Variable bins_sum(const Variable &data);
+[[nodiscard]] SCIPP_VARIABLE_EXPORT Variable bins_nansum(const Variable &data);
+[[nodiscard]] SCIPP_VARIABLE_EXPORT Variable bins_max(const Variable &data);
+[[nodiscard]] SCIPP_VARIABLE_EXPORT Variable bins_nanmax(const Variable &data);
+[[nodiscard]] SCIPP_VARIABLE_EXPORT Variable bins_min(const Variable &data);
+[[nodiscard]] SCIPP_VARIABLE_EXPORT Variable bins_nanmin(const Variable &data);
+[[nodiscard]] SCIPP_VARIABLE_EXPORT Variable bins_all(const Variable &data);
+[[nodiscard]] SCIPP_VARIABLE_EXPORT Variable bins_any(const Variable &data);
+[[nodiscard]] SCIPP_VARIABLE_EXPORT Variable bins_mean(const Variable &data);
+[[nodiscard]] SCIPP_VARIABLE_EXPORT Variable bins_nanmean(const Variable &data);
+
 // These reductions accumulate their results in their first argument
 // without erasing its current contents.
 SCIPP_VARIABLE_EXPORT void sum_into(Variable &accum, const Variable &var);
+SCIPP_VARIABLE_EXPORT void nansum_into(Variable &accum, const Variable &var);
 SCIPP_VARIABLE_EXPORT void all_into(Variable &accum, const Variable &var);
 SCIPP_VARIABLE_EXPORT void any_into(Variable &accum, const Variable &var);
 SCIPP_VARIABLE_EXPORT void max_into(Variable &accum, const Variable &var);
+SCIPP_VARIABLE_EXPORT void nanmax_into(Variable &accum, const Variable &var);
 SCIPP_VARIABLE_EXPORT void min_into(Variable &accum, const Variable &var);
-
+SCIPP_VARIABLE_EXPORT void nanmin_into(Variable &accum, const Variable &var);
```

Comment:
Now that we have these, we should be able to trivially add the `nan*` variants to `groupby`?